internal/teeingfile.go

```diff
@@ -0,0 +1,94 @@
+package internal
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"sync"
+)
+
+type TeeingFile struct {
+	ctx     context.Context
+	lk      sync.Mutex
+	file    *os.File
+	tee     bytes.Buffer
+	closeCh chan struct{}
+
+	offset int64
+}
+
+func NewTeeingFile(ctx context.Context, file *os.File) *TeeingFile {
+	return &TeeingFile{
+		ctx:     ctx,
+		file:    file,
+		closeCh: make(chan struct{}),
+	}
+}
+
+func (t *TeeingFile) WriteTo(out io.Writer) error {
+	errCh := make(chan error)
+	go func() {
```

Comment:
don't look too closely at this .. it needs to be replaced with something with better signalling, probably some back-pressure so we don't end up bloating memory if the receiver is slow, which might mean an `io.Pipe`. For now it's just a spinning loop that may or may not end properly.